src/lib/pages/backoffice/Actions/CheckOut/state/actions.js

```diff
@@ -0,0 +1,119 @@
+import { itemApi } from '@api/items';
+import { patronApi } from '@api/patrons';
+import {
+  addNotification,
+  sendErrorNotification,
+} from '@components/Notifications';
+import { invenioConfig } from '@config';
+import { goTo } from '@history';
+import { BackOfficeRoutes } from '@routes/urls';
+
+export const CLEAR_SEARCH = 'checkOutSearch/CLEAR_SEARCH';
+export const CLEAR_RESULTS = 'checkOutSearch/CLEAR_RESULTS';
+export const QUERY_STRING_UPDATE = 'checkOutSearch/QUERY_STRING_UPDATE';
+export const SEARCH_IS_LOADING = 'checkOutSearch/SEARCH_IS_LOADING';
+export const SEARCH_HAS_ERROR = 'checkOutSearch/SEARCH_HAS_ERROR';
+export const SEARCH_PATRON_SUCCESS = 'checkOutSearch/SEARCH_PATRON_SUCCESS  ';
+export const SEARCH_ITEM_SUCCESS = 'checkOutSearch/SEARCH_ITEM_SUCCESS  ';
+
+export const updateQueryString = queryString => {
+  return dispatch => {
+    dispatch({
+      type: QUERY_STRING_UPDATE,
+      payload: queryString,
+    });
+  };
+};
+
+export const clearSearch = () => {
+  return dispatch => {
+    dispatch({
+      type: CLEAR_SEARCH,
+    });
+  };
+};
+
+export const clearResults = () => {
+  return dispatch => {
+    dispatch({
+      type: CLEAR_RESULTS,
+    });
+  };
+};
+
+const searchPatrons = async (dispatch, term) => {
+  const response = await patronApi.list(
+    patronApi
+      .query()
+      .withEmail(term, true)
+      .withName(term, true)
+      .withPatronUniqueID(term)
```

Comment:
This should be something like: `withCustomField`(s) or something similar, since this is configurable.